### PR TITLE
PP-to-Cube time rounding.

### PR DESCRIPTION
**Pull request by rhattersley**
Floating point errors can creep in when converting PP header datetimes to hours-since-1970:
```python
>>> iris.unit.Unit('hours since epoch').date2num(netcdftime.datetime(2015, 1, 20, 7, 0))
394926.9999999963
```
This can have a corresponding knock-on effect on calculated forecast periods, and downstream processing.

But PP header datetimes have a maximum precision of one second, so when expressed as hours we can safely round off to a few decimal places (sufficient to capture 1 / 3600).

### CODE REVIEW

File: lib/iris/fileformats/pp_rules.py [hunk 1]
```diff
@@ -423,15 +423,16 @@ def _convert_time_coords(lbcode, lbtim, epoch_hours_unit,
...
-            np.atleast_1d(times_array)).reshape(times_array.shape)
+    def date2hours(t):
+        epoch_hours = epoch_hours_unit.date2num(t)
+        if t.minute == 0 and t.second == 0:
+            epoch_hours = round(epoch_hours)
+        return epoch_hours
 
-    t1_epoch_hours = date2hours(t1)
-    t2_epoch_hours = date2hours(t2)
+    dates2hours = np.vectorize(date2hours)
```
Comment (pelson) [inline lib/iris/fileformats/pp_rules.py]: I guess this is going to impact the performance of experimental structured loading, but how much I'm not sure...

**Comment by bjlittle** [comment]
@rhattersley Travis is complaining ... your fix is rounding a whole bunch of the test results.

**Comment by rhattersley** [comment]
> your fix is rounding a whole bunch of the test results.

Thanks for the reminder. I fixed a bunch of them last night, but the remaining changes I want to keep. I'll update...

**Comment by rhattersley** [comment]
@bjlittle - now passing with updated test results.

**Comment by bjlittle** [comment]
:+1:

**Comment by bjlittle** [comment]
@rhattersley ... hang fire ... just realised that your targeting `v1.8.x`

We're aiming to cut a release today, are we happy to include this even though it's not been part of the pre-release? Or should we cut the release without this, and include it as a bug-fix aka `v1.8.1` just incase? 

Thoughts ...

**Comment by rhattersley** [comment]
> just realised that your targeting v1.8.x

Yeah, I'm aiming for 1.8.1

**Comment by bjlittle** [comment]
@rhattersley Great, thanks for clarifying :+1:

**Comment by rhattersley** [comment]
With 1.8.0 out the door we can think about merging this again.

**Pull request merged**